### ISSUE TITLE
create custom columns for user->site and user->cashdrawer settings for el dorado

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#1994]
_Imported from trac ticket 1994.
Created by **mariam**
Opened in trac: Oct 25, 2012, 10:56:39 AM
Last modified in trac: Aug 28, 2013, 1:51:28 PM_
***
_by mariam:_
I don't know if Mariam has brought up this before, but we are having an issue with our thin-clients and PCs slowing down tremendously after extended periods of using LOFT.   We found that the way to make LOFT run quickly again is to clear the Cache on the clients.  If not done correctly, the cookies also get cleared.

This results in having the salespeople deciding what their store number and cashier number is when logging into LOFT for the first time.  This also happens every time we get a new update.  As a result, we are having people entering the wrong store number and wrong cashier station numbers.  This will only get worse once we begin using credit cards with LOFT, since the cashier station will be MUCH MORE important in order to balance the money taken by the salespeople.

Bottom line:  Can we tie the salesperson to the store/cashier station?  This could be as an OPTION. Have each company decide whether to tie the physical station or the person to the store/cash drawer.

___

Use model customizations to handle this and then add a rule hook in the root controller's authenticate method to allow for extra checks at login where we can use the new data elements.

**Comment by kentbower** [comment]
_**rneville** trac comment on Jan 11, 2013, 10:03:54 AM:_

pulling forward

**Comment by kentbower** [comment]
_**rneville** trac comment on Feb 8, 2013, 10:48:27 AM:_

moving forward to sprint 23

**Comment by kentbower** [comment]
_**rneville** trac comment on Aug 28, 2013, 1:51:28 PM:_

I started on this and then hit a wall when I realized that this doesn't work as a rule.... site id and cashdrawer are not stored in a persistent "session" on the server. The "session", if there is one, is stored in memory on the client. In fact, the authenticate service doesn't even get the site and cashdrawer passed to it. Instead, those values are store client-side and sent as arguments to appropriate service calls that need that data.

The only way we could do this is to a) implement it in the client which means b) it would have to be a global 'feature' for everyone - which I'm not sure we want to do.
